framework/doc/content/source/postprocessors/SideAdvectiveFluxIntegral.md

```diff
@@ -0,0 +1,17 @@
+# SideAdvectiveFluxIntegral
+
+!syntax description /Postprocessors/SideAdvectiveFluxIntegral
+
+## Description
+
+The `SideAdvectiveFluxIntegral` postprocessor is used to compute the side integral of the advection flux component including x, y, z and normal. This kernel supports both finite element and finite volume variables. For finite element method, the variable needs to be LAGRANGE, so that the side value are well-defined. The advection variable can be either a variable[!param](/Postprocessors/SideAdvectiveFluxIntegral/advected_variable) or material property[!param](/Postprocessors/SideAdvectiveFluxIntegral/advected_mat_prop). For finite volume method, an advected quantity needs to be provided[!param](/Postprocessors/SideAdvectiveFluxIntegral/advected_quantity). The advection velocity is required with [!param](/Postprocessors/SideAdvectiveFluxIntegral/vel_x) for advection flux calculation. [!param](/Postprocessors/SideAdvectiveFluxIntegral/vel_y) and [!param](/Postprocessors/SideAdvectiveFluxIntegral/vel_z) are needed for 2D and 3D simulations.
```

Comment:
why should it be lagrange? most families will have a known side value, as long as you specify which side you're coming from

Reply:
For MONOMIAL, the side value will be all 0. Even the side is specified, there may be discontinuous on the side, therefore MONOMIAL doesn't has a defined side value. The side value is only defined in Finite Volume.

Reply:
yeah but what about higher order "elemental" variables?

Reply:
I have tried first order monomial, the results are the same, which I think makes sense.

Reply:
Any continuous FE type will give you a well-defined-from-both-sides value on a side.  That includes `LAGRANGE`, `HIERARCHIC`, `BERNSTEIN`, `SZABAB`, `RATIONAL_BERNSTEIN`, `CLOUGH`, `HERMITE`...

Reply:
And for *advective* flux, in particular, you don't generally need the value to be well-defined from both sides, do you?  Where the advection vector is tangential to the side, the advective flux is zero; where the advection vector has a normal component you want to evaluate the advected value from the upstream element.

Reply:
"Return the flux calculation using the upstream element" might be a bit of extra code that we don't want to write right now, but in that case the thing to do would be "throw an error", not "return 0".